tests/unit/KeyedObjectsContainers.cpp

```diff
@@ -201,3 +202,38 @@ BOOST_AUTO_TEST_CASE(TestUniqueLast) {
   Vector Expected = { { 1, 3 }, { 2, 3 } };
   revng_check(TheVector == Expected);
 }
+
+BOOST_AUTO_TEST_CASE(TrackingContainerVectorAt) {
+  revng::TrackingContainer<SortedVector<int>> Vector;
+  int One(1);
+  Vector.insert(One);
+  Vector.resetTracking();
+  const auto &Reference = Vector;
+  Reference.at(One);
+
+  revng_check(Reference.getExistingRequestedKeys().size() == 1);
+  revng_check(Reference.getExistingRequestedKeys().top().front() == 1);
+}
+
+BOOST_AUTO_TEST_CASE(TrackingContainerVectorCount) {
+  revng::TrackingContainer<SortedVector<int>> Vector;
+  Vector.insert(1);
+  Vector.resetTracking();
+  const auto &Reference = Vector;
+  Reference.count(2);
+
+  revng_check(Reference.getExistingRequestedKeys().top().size() == 0);
+  revng_check(Reference.getAccessCounter().isSet());
+}
+
+BOOST_AUTO_TEST_CASE(TrackingContainerSetCount) {
+  revng::TrackingContainer<MutableSet<int>> Vector;
```

Comment:
* More testing
* Add these to `testSet` or make a new function that works both with `SortedVector` and `MutableSet` so you can write code once and test both.

Reply:
are you sure you want to generate this kinds of tests? nothing in the impelmentation of trackingContainerSet relies on things that are specific to the set itself. Templetizing the tests would just make it harder for people to reproduce them, should they break one by accident.

Reply:
> nothing in the impelmentation of trackingContainerSet relies on things that are specific to the set itself

Well, this claim needs testing! `:)` Unfortunately we cannot assert that `TrackingContainer<T>` only uses methods made available by the concept constraint on `T`, and even if it could, we might be missing runtime failures.
It's not too bad, you end up in `gdb` and you have to take a look at the name of the template specialization. It happened to me recently.